src/libcrun/linux.c

```diff
@@ -3192,6 +3192,13 @@ validate_sysctl (const char *original_value, const char *name, unsigned long nam
 
       if (strcmp (name, "kernel/domainname") == 0)
         {
+          // Value of sysctl `kernel/domainname` is going to
+          // conflict with already set field `domainname` in
+          // OCI spec, in such scenario crun will fail to prevent
+          // unexpected behaviour for end user.
+          if (! is_empty_string (def->domainname) && (strcmp (original_value, def->domainname) != 0))
```

Comment:
could we use `def->domainname_present` to check for the presence of the domainname definition?

Reply:
@giuseppe I am not at all sure but could you please confirm my finding? I think `_present` is only generated for nested structs such as `config_linux` or `config_solaris` etc, `domainname` and `hostname` are the part of top level config, I think `libocispec` needs to be patched to supported this ( sounds like a good feature add-on ). 

But currently I am not at all sure if `def->domainanme_present` can be used as is in the current version of `libocispec`.

Reply:
it is not blocking, I am fine with the current version.  Thanks for checking it!